src/order/filter/n_ary.lean

```diff
@@ -49,6 +50,38 @@ def map₂ (m : α → β → γ) (f : filter α) (g : filter β) : filter γ :=
 lemma image2_mem_map₂ (hs : s ∈ f) (ht : t ∈ g) : image2 m s t ∈ map₂ m f g :=
 ⟨_, _, hs, ht, subset.rfl⟩
 
+lemma map_prod_eq_map₂ {f : filter α} {g : filter β} (m : α → β → γ) :
+  filter.map (λ p : α × β, m p.1 p.2) (f ×ᶠ g) = map₂ m f g :=
+begin
+  ext s,
+  split,
+  { intro hmem,
+    rw mem_map₂_iff,
+    rw filter.mem_map_iff_exists_image at hmem,
+    obtain ⟨s', hs', hsub⟩ := hmem,
+    rw filter.mem_prod_iff at hs',
+    obtain ⟨t, ht, t', ht', hsub'⟩ := hs',
+    refine ⟨t, t', ht, ht', _⟩,
+    rw ← set.image_prod,
+    exact subset_trans (set.image_subset (λ (p : α × β), m p.fst p.snd) hsub') hsub },
+  { intro hmem,
+    rw mem_map₂_iff at hmem,
+    obtain ⟨t, t', ht, ht', hsub⟩ := hmem,
+    rw ← set.image_prod at hsub,
+    rw filter.mem_map_iff_exists_image,
+    exact ⟨t ×ˢ t', filter.prod_mem_prod ht ht', hsub⟩ },
+end
+
+lemma map_prod_eq_map₂' {f : filter α} {g : filter β} (m : α × β → γ) :
+  filter.map m (f ×ᶠ g) = map₂ (λ a b, m (a, b)) f g :=
+begin
+  have heq := map_prod_eq_map₂ (curry m),
```

Comment:
I would guess
```suggestion
  refine eq.trans _ map_prod_eq_map₂ (curry m),
```
Works here

Reply:
Nice. I was able to simplify the proof to a one liner.